Source/Core/DolphinQt/Config/GameConfigEdit.cpp

```diff
@@ -0,0 +1,170 @@
+  QFile file(m_path);
+  if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+    return;
+
+  setPlainText(QString::fromStdString(file.readAll().toStdString()));
+}
+
+#include <iostream>
+
+void GameConfigEdit::SaveFile()
+{
+  if (!isVisible() || isReadOnly())
+    return;
+
+  QFile file(m_path);
+
+  if (!file.open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text))
+    return;
+
+  const std::string contents = toPlainText().toStdString();
+
+  file.write(contents.c_str(), contents.length());
+}
+
+void GameConfigEdit::Connect()
+{
+  connect(this, &QTextEdit::textChanged, this, &GameConfigEdit::SaveFile);
+  connect(this, &QTextEdit::selectionChanged, this, &GameConfigEdit::OnSelectionChanged);
+  connect(this, &QTextEdit::customContextMenuRequested, this, &GameConfigEdit::OnContextMenu);
+}
+
+void GameConfigEdit::OnSelectionChanged()
+{
+  const QString& keyword = textCursor().selectedText();
+
+  if (m_keyword_map.count(keyword))
+    QWhatsThis::showText(QCursor::pos(), m_keyword_map[keyword], this);
+}
+
+void GameConfigEdit::AddBoolOption(QMenu* menu, const QString& name, const QString& section,
+                                   const QString& key)
+{
+  auto* option = menu->addMenu(name);
+
+  option->addAction(tr("On"), this,
+                    [this, section, key] { SetOption(section, key, QStringLiteral("True")); });
+  option->addAction(tr("Off"), this,
+                    [this, section, key] { SetOption(section, key, QStringLiteral("False")); });
+}
+
+void GameConfigEdit::SetOption(const QString& section, const QString& key, const QString& value)
+{
+  auto section_cursor = document()->find(QRegExp(QStringLiteral("^\\[%1\\]").arg(section)), 0);
+
+  // Check if the section this belongs in can be found
+  if (!section_cursor.isNull())
+  {
+    auto value_cursor =
+        document()->find(QRegExp(QStringLiteral("^%1 = .*").arg(key)), section_cursor);
+
+    const QString new_line = QStringLiteral("%1 = %2").arg(key).arg(value);
+
+    // Check if the value that has to be set already exists
+    if (!value_cursor.isNull())
+    {
+      value_cursor.insertText(new_line);
+    }
+    else
+    {
+      section_cursor.clearSelection();
+      section_cursor.insertText(QStringLiteral("\n") + new_line);
+    }
+  }
+  else
+  {
+    append(QStringLiteral("[%1]\n\n%2 = %3\n").arg(section).arg(key).arg(value));
+  }
+}
+
+void GameConfigEdit::OnContextMenu(const QPoint& pos)
+{
+  auto* menu = createStandardContextMenu(pos);
+
+  menu->addSeparator();
+
+  menu->addAction(tr("Refresh"), this, &GameConfigEdit::LoadFile);
+  menu->addAction(tr("Open in External Editor"), this, &GameConfigEdit::OpenExternalEditor);
+
+  if (!isReadOnly())
+  {
+    menu->addSeparator();
+    auto* core_menu = menu->addMenu(tr("Core"));
+
+    AddBoolOption(core_menu, tr("Dual-Core"), QStringLiteral("Core"), QStringLiteral("CPUThread"));
+    AddBoolOption(core_menu, tr("Fastmem"), QStringLiteral("Core"), QStringLiteral("Fastmem"));
```

Comment:
Why are we exposing fastmem in particular? A normal user shouldn't have to edit it, and I don't see any reason for anyone ever to set it on a per-game basis.

Also, "Fastmem" sounds too much like developer jargon for a translatable string. I can imagine that translators will have trouble coming up with a translation that gives off the same feel while at the same time not sounding weird.

Reply:
See it more as a placeholder.